components/nodemanager-service/nodemanager.go

```diff
@@ -103,14 +135,165 @@ func serve(ctx context.Context, config *config.Nodemanager, connFactory *securec
+		if err != nil {
+			return errors.Wrapf(err, "failed to register workflow for %q", jobName)
+		}
+	}
+
+	// SOON: Make all of *waves hands* less repetitive.
+	// Prelude to initializing the manager: create rrules for schedule
+	// Right now, this is migration only -- doesn't handle config change.
+	awsPollInterval := config.Manager.AwsEc2PollIntervalMinutes
+	rule, err := rrule.NewRRule(rrule.ROption{
+		Freq:     rrule.MINUTELY,
+		Interval: awsPollInterval,
+		Dtstart:  time.Now(),
+	})
+	if err != nil {
+		return errors.Wrapf(err, "Could not create recurrence rule for nodemanager workflow %s", Awsec2PollingJobName)
+	}
+
+	azurePollInterval := config.Manager.AzureVMPollIntervalMinutes
+	azureRule, err := rrule.NewRRule(rrule.ROption{
+		Freq:     rrule.MINUTELY,
+		Interval: azurePollInterval,
+		Dtstart:  time.Now(),
+	})
+	if err != nil {
+		return errors.Wrapf(err, "Could not create recurrence rule for nodemanager workflow %s", AzurevmPollingJobName)
+	}
+
+	// managersPollInterval := 120
+	// poll faster for debugging
+	managersPollInterval := 1
+	managersRule, err := rrule.NewRRule(rrule.ROption{
+		Freq:     rrule.MINUTELY,
+		Interval: managersPollInterval,
+		Dtstart:  time.Now(),
+	})
+	if err != nil {
+		return errors.Wrapf(err, "Could not create recurrence rule for nodemanager workflow %s", ManagersPollingJobName)
+	}
+
+	err = cerealManager.CreateWorkflowSchedule(Awsec2PollingScheduleName, Awsec2PollingJobName, nil, true, rule)
+	if err != nil {
+		if err == cereal.ErrWorkflowScheduleExists {
+			log.Infof("nodemanager workflow schedule %s already exists, not migrating", Awsec2PollingScheduleName)
+			// If the schedule exists, make sure the rrule is up-to-date.
+			schedule, err := cerealManager.GetWorkflowScheduleByName(ctx, Awsec2PollingScheduleName, Awsec2PollingJobName)
+			if err != nil {
+				return errors.Wrapf(err, "failed to get scheduled workflow %s from cereal manager", Awsec2PollingScheduleName)
+			}
+			scheduledRule, err := schedule.GetRRule()
+			if err != nil {
+				return errors.Wrapf(err, "unable to get rrule for scheduled workflow %s", Awsec2PollingScheduleName)
+			}
+			if scheduledRule != rule {
+				err = cerealManager.UpdateWorkflowScheduleByName(context.Background(), Awsec2PollingScheduleName, Awsec2PollingJobName, cereal.UpdateRecurrence(rule))
+				if err != nil {
+					return errors.Wrapf(err, "unable to update recurrence rule for scheduled workflow %s", Awsec2PollingScheduleName)
+				}
+			}
+		} else {
+			return errors.Wrapf(err, "Could not continue creating workflow schedule %s", Awsec2PollingScheduleName)
+		}
+	}
+
+	err = cerealManager.CreateWorkflowSchedule(AzurevmPollingScheduleName, AzurevmPollingJobName, nil, true, azureRule)
+	if err != nil {
+		if err == cereal.ErrWorkflowScheduleExists {
+			log.Infof("nodemanager workflow schedule %s already exists, not migrating", AzurevmPollingScheduleName)
```

Comment:
I'm not sure you're migrating anything

Reply:
Yes, I need to rewrite this error message. We're still using the config to set the various polling intervals, just changing the mechanism that consume them.